chrome/content/zotero/tools/csledit.js

```diff
@@ -69,13 +61,30 @@ var Zotero_CSL_Editor = new function() {
 		}
 		
 		pageList.selectedIndex = 0;
-		cslList.setAttribute('initialized', true);
 	}
-	function refresh() {
-		var editor = document.getElementById('zotero-csl-editor');
-		generateBibliography(editor.value);
-
+	
+	this.onStyleSelected = function(styleID) {
+		window.setTimeout(
```

Comment:
> The point here is that updateLocaleList cannot be called synchronously after populateLocaleList, hence the timeout.

Why?

> But secondly, the timeout is really only necessary for the call from init, so that's where this should have been.

Why?

(sorry, rather new to the async stuff)